### PR TITLE
Bring comments in profile/replies up to feature parity with posts

### PR DESCRIPTION
**Pull request by CTalvio**
## Pull Request Description

Significant refactoring of comment widgets to allow the same files to be used in posts, profiles, and replies.

Mentions remain unchanged, as they can include both posts and comments and will hence need more work.

## Issue Being Fixed

- #220
- #362 

## Screenshots / Recordings

image
image

## Checklist

- [x] Did you update CHANGELOG.md?

### CODE REVIEW

**Comment by CTalvio** [comment]
Screenshot_20230823_044303

@micahmo @ajsosa We should get this merged. It's not entirely functional, but it breaks nothing of what is already in, and  it adds a lot for us to build off of. Here's what would need to be done in additional PRs:
- Favoriting from inbox works, but does not update view
- Voting from inbox does not work
- Editing/replying from profile, does not work
- Long press in either does not bring up comment modal

Aside from this, its all good. I'll be around to fix anything you need for a merge, but after that I'll likely take a break from work on Thunder. I'll be back, tho.

File: pubspec.lock [hunk 1]
```diff
@@ -1227,10 +1227,10 @@ packages:
     dependency: "direct main"
     description:
       name: webview_flutter
-      sha256: "789d52bd789373cc1e100fb634af2127e86c99cf9abde09499743270c5de8d00"
+      sha256: "04a0782fb058b7c71f2048935583488f4d32e9147ca403abc4e58f1de9964629"
       url: "https://pub.dev"
     source: hosted
-    version: "4.2.2"
+    version: "4.2.3"
```
Comment (micahmo) [inline pubspec.lock:1233]: Is this change intentional?

File: lib/post/widgets/create_comment_modal.dart [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (micahmo) [inline lib/post/widgets/create_comment_modal.dart:1]: Is this a new, empty file?

**Comment by CTalvio** [comment]
Will resolve those comments later today

File: pubspec.lock (hunk 1 above)
Comment (CTalvio) [reply pubspec.lock:1233]: No

File: lib/shared/comment_reference.dart [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+                          ],
+                        ),
+                      ],
+                    ),
+                  ),
+                  if (widget.child != null) widget.child!,
+                ],
+              ),
+            ),
+            /*const Divider(height: 1),*/
```
[the remark this answers is not in the document]
Comment (CTalvio) [reply lib/shared/comment_reference.dart:163]: Looks like no, IIRC I used a divider in a earlier design for the comment reference list item. Removed it.

**Comment by CTalvio** [comment]
Done

**Comment by ajsosa** [comment]
Looks good to me 👍
